blaze/compute/sparksql.py

```diff
@@ -1,135 +1,136 @@
 from __future__ import absolute_import, division, print_function
 
-import toolz
-from toolz import pipe
-import itertools
-from datashape import discover, Unit, Tuple, Record, iscollection, isscalar
+import datashape
+from datashape import discover
+from datashape.predicates import isrecord
 import sqlalchemy as sa
+from toolz import pipe, curry
+from toolz.curried import filter
+from into import convert
 from into.backends.sql import dshape_to_alchemy
 
 from ..dispatch import dispatch
-from ..expr import *
+from ..expr import Expr, Field, symbol
+from .core import compute
 from .utils import literalquery
 
-__all__ = []
-
-try:
-    import pyspark
-    from pyspark.sql import SchemaRDD
-except ImportError:
-    SchemaRDD = type(None)
-
-names = ('_table_%d' % i for i in itertools.count(1))
 
 __all__ = []
 
-class SparkSQLQuery(object):
-    """ Pair of PySpark SQLContext and SQLAlchemy Table
-
-    Python's SparkSQL interface only accepts strings.  We use SQLAlchemy to
-    generate these strings.  To do this we'll have to pass around pairs of
-    (SQLContext, sqlalchemy.Selectable).  Additionally we track a mapping of
-    {schemardd: sqlalchemy.Table}
-
-    Parameters
-    ----------
-
-    context: pyspark.sql.SQLContext
-
-    query: sqlalchemy.Selectable
-
-    mapping: dict :: {pyspark.sql.SchemaRDD: sqlalchemy.Table}
-    """
-    __slots__ = 'context', 'query', 'mapping'
+from pyspark.sql import SQLContext
+import pyspark.sql as sql
 
-    def __init__(self, context, query, mapping):
-        self.context = context
-        self.query = query
-        self.mapping = mapping
 
+@convert.register(list, sql.DataFrame)
+def sparksql_dataframe_to_list(df, **kwargs):
+    return list(map(tuple, df.collect()))
 
-def make_query(rdd, primary_key='', name=None):
-    # SparkSQL
-    name = name or next(names)
-    context = rdd.sql_ctx
-    context.registerRDDAsTable(rdd, name)
 
-    # SQLAlchemy
-    schema = discover(rdd).subshape[0]
-    columns = dshape_to_alchemy(schema)
-    for column in columns:
-        if column.name == primary_key:
-            column.primary_key = True
+@dispatch(Field, SQLContext)
+def compute_up(expr, data, **kwargs):
+    name = expr._name
+    columns = dshape_to_alchemy(expr.dshape)
+    return sa.Table(name, sa.MetaData(), *columns)
```

Comment:
In this particular case it seems like we're not actually using this in the normal way of `compute_up`.  Could we call it something else and still have the code work the same?

The thing that it returns isn't really the expression mapped on to the data.  We wouldn't be able to recover an actual SparkSQL dataframe here.